test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs

```diff
@@ -292,7 +292,7 @@ public async Task FindsReferencesOfVariable()
             Assert.Equal(12, locations[2].Range.Start.Character);
         }
 
-        [Fact]
+        [Fact(Skip = "AppVeyor is currently hanging on this test it seems")]
```

Comment:
is this test the culprit?

Reply:
It *seems* to be. It's strange. *Just* skipping this test doesn't work, and *just* putting timeouts on the `async` tests doesn't work. But both together passes, consistently. Naturally, "if you fix it but you don't know how, you didn't fix it"...

For now I just want to get tests running and passing in CI, but I think this is another signal that reworking the tests is needed

Reply:
I wonder if the timeout isn't doing anything for some tests because there isn't an `await` soon enough. From what I understand async methods are ran on the same thread they are started until they hit the first `await` keyword.  There may be something that *appears* to be async but doesn't actually use the keyword enough.

If you want you could try forcing the test into another thread by running it with `Task.Run`

Reply:
It's back to hanging again :angry:

Reply:
The initial invocation of the task needs to be in the `Task.Run` block.  My guess is that the method never actually runs a task in the first place, it just hangs before it hits an `await`.

If you wanted to make it a utility method you could do something like

```csharp
async Task<T> RunWithTimeout<T>(Func<Task<T>> test, int timeout)
{
    ((IAsyncResult)Task.Run(test)).WaitOne(..etc
```

Reply:
Bugger, still hanging

Reply:
Ok, reverting back to the last commit that passed this passes again... :unamused:

Reply:
I'm just going to merge this for now so the other PRs can test properly

Reply:
>  async methods are ran on the same thread they are started until they hit the first await keyword.

Even then, IIRC an async operation can complete during the initial awaited async call if the operation determines it can return the result right away.